mmcv/runner/fp16_utils.py

```diff
@@ -233,14 +235,38 @@ def allreduce_grads(params, coalesce=True, bucket_size_mb=-1):
         param.grad.data for param in params
         if param.requires_grad and param.grad is not None
     ]
-    world_size = dist.get_world_size()
+    _, world_size = get_dist_info()
+    if world_size == 1:
+        return
     if coalesce:
         _allreduce_coalesced(grads, world_size, bucket_size_mb)
     else:
         for tensor in grads:
             dist.all_reduce(tensor.div_(world_size))
 
 
+def allreduce_params(params, coalesce=True, bucket_size_mb=-1):
```

Comment:
It is not only designed for fp16, thus we can put it in `dist_utils.py` or `utils.py`.